src/document/json/array.ts

```diff
@@ -85,16 +90,20 @@ export class JSONArray extends JSONContainer {
     }
   }
 
-  public *getDescendants(): IterableIterator<JSONElement> {
+  public getDescendants(
+    callback: (elem: JSONElement, parent: JSONContainer) => boolean,
+  ): void {
     for (const node of this.elements) {
       const element = node.getValue();
-      if (element instanceof JSONContainer) {
-        for (const descendant of element.getDescendants()) {
-          yield descendant;
-        }
+      if (callback(element, this)) {
+        return;
       }
 
-      yield element;
+      if (element instanceof JSONObject) {
+        element.getDescendants(callback);
+      } else if (element instanceof JSONArray) {
+        element.getDescendants(callback);
+      }
```

Comment:
instead of this

```javascript
element instanceof JSONObject || element instanceof JSONArray
```
, How about changing like this? 🤔

```javascript
if (element instanceof JSONContainer) {
   element.getDescendants(callback);
}
```

Reply:
I think the way of comparing `element` with `JSONContainer` is more readable. 👍